internal/tasks/schedulertask_test.go

```diff
@@ -140,3 +143,189 @@ func TestSchedulerTask_Run(t *testing.T) {
+		{
+			Name:               "projects/test-project/topics/test",
+			NumPartitions:      2,
+			ExtractedTopicName: "test",
+			ExtractedProjectID: "test-project",
+			Labels:             map[string]string{},
+		},
+	}
+}
+
+func GetDummySubModel() subscription.Model {
+	return subscription.Model{
+		Name:                           "projects/test-project/subscriptions/test",
+		Topic:                          "projects/test-project/topics/test",
+		ExtractedTopicProjectID:        "test-project",
+		ExtractedSubscriptionName:      "test",
+		ExtractedSubscriptionProjectID: "test-project",
+		ExtractedTopicName:             "test",
+		DeadLetterPolicy: &subscription.DeadLetterPolicy{
+			DeadLetterTopic:     "projects/test-project/topics/test-dlq",
+			MaxDeliveryAttempts: 5,
+		},
+		PushConfig: &subscription.PushConfig{
+			PushEndpoint: "http://test.test",
+		},
+	}
+}
+
+func TestSchedulerTask_rebalanceSubs(t *testing.T) {
+	type fields struct {
+		id               string
+		registry         registry.IRegistry
+		brokerstore      brokerstore.IBrokerStore
+		nodeCore         node.ICore
+		scheduler        scheduler.IScheduler
+		topicCore        topic.ICore
+		nodeBindingCore  nodebinding.ICore
+		subscriptionCore subscription.ICore
+		nodeCache        map[string]*node.Model
+		subCache         map[string]*subscription.Model
+		topicCache       map[string]*topic.Model
+		nodeWatchData    chan *struct{}
+		subWatchData     chan *struct{}
+		topicWatchData   chan *struct{}
+	}
+	type args struct {
+		ctx context.Context
+	}
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	registryMock := mocks.NewMockIRegistry(ctrl)
+	brokerstoreMock := mocks2.NewMockIBrokerStore(ctrl)
+	nodeCoreMock := mocks3.NewMockICore(ctrl)
+	topicCoreMock := mocks4.NewMockICore(ctrl)
+	subscriptionCoreMock := mocks7.NewMockICore(ctrl)
+	nodebindingCoreMock := mocks5.NewMockICore(ctrl)
+	schedulerMock := mocks6.NewMockIScheduler(ctrl)
+
+	workerID := uuid.New().String()
+	workerID2 := uuid.New().String()
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Rebalance subs failed",
+			fields: fields{
+				id:               workerID,
+				registry:         registryMock,
+				brokerstore:      brokerstoreMock,
+				nodeCore:         nodeCoreMock,
+				scheduler:        schedulerMock,
+				topicCore:        topicCoreMock,
+				nodeBindingCore:  nodebindingCoreMock,
+				subscriptionCore: subscriptionCoreMock,
+				nodeCache:        make(map[string]*node.Model),
+				subCache:         make(map[string]*subscription.Model),
+				topicCache:       make(map[string]*topic.Model),
+				nodeWatchData:    make(chan *struct{}),
+				subWatchData:     make(chan *struct{}),
+				topicWatchData:   make(chan *struct{}),
+			},
+			args: args{
+				ctx: ctx,
+			},
+			wantErr: true,
+		},
+		{
+			name: "Rebalance subs successfully",
```

Comment:
How are we testing that the rebalance logic is being applied and no single node has greater than median nodebindings?

Reply:
I have added test cases for 3 node ids, where 2 of the nodes have less than median node bindings.